packages/addons/service/librespot/source/default.py

```diff
@@ -79,6 +79,7 @@ def stop(self):
 class Player(xbmc.Player):
 
    ITEM = 'rtp://127.0.0.1:{port}'.format(port=PORT)
+   RESTART_ON_STOP = True
```

Comment:
This changes the current behaviour of the addon, and requires editing a file to revert.
Therefore, either remove this change or introduce it through an addon setting

Reply:
Hm... I read too fast.
@Obrekr can you clarify what this does?

Reply:
I think the behaviour of librespot changed at some point concerning the events being send, here an example of generated events:

    May 20 12:22:06 Dorayaki systemd[1]: Started librespot.
    May 20 12:22:06 Dorayaki sh[6979]: INFO:librespot: librespot 85b8697 (2018-05-19). Built on 2018-05-19. Build ID: j20C7h5W
    May 20 12:22:06 Dorayaki sh[6979]: *** WARNING *** The program 'librespot' uses the Apple Bonjour compatibility layer of Avahi.
    May 20 12:22:06 Dorayaki librespot[6994]: *** WARNING *** The program 'librespot' uses the Apple Bonjour compatibility layer of Avahi.
    May 20 12:22:06 Dorayaki librespot[6994]: *** WARNING *** Please fix your application to use the native API of Avahi!
    May 20 12:22:06 Dorayaki sh[6979]: *** WARNING *** Please fix your application to use the native API of Avahi!
    May 20 12:22:06 Dorayaki sh[6979]: *** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
    May 20 12:22:06 Dorayaki librespot[6994]: *** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
    May 20 12:22:06 Dorayaki sh[6979]: INFO:librespot_core::session: Connecting to AP "gew1-accesspoint-b-jkm8.ap.spotify.com:4070"
    May 20 12:22:06 Dorayaki sh[6979]: INFO:librespot_core::session: Authenticated as "obrekr" !
    May 20 12:22:06 Dorayaki sh[6979]: INFO:librespot_core::session: Country: "DE"
    # connecting to librespot
    May 20 12:22:11 Dorayaki sh[6979]: INFO:librespot_playback::player: Loading track "Tainted" with Spotify URI "spotify:track:03noobdJp6S95MRoYDXNQF"
    May 20 12:22:12 Dorayaki sh[6979]: INFO:librespot_playback::player: Track "Tainted" loaded
    May 20 12:22:12 Dorayaki sh[6979]: INFO:librespot::player_event_handler: Running ["librespot.onevent"] with environment variables {"PLAYER_EVENT": "stop", "TRACK_ID": "01ca309fd73f4e15ba835c39423be41d"}
    # hitting play in Spotify
    May 20 12:22:13 Dorayaki sh[6979]: INFO:librespot::player_event_handler: Running ["librespot.onevent"] with environment variables {"TRACK_ID": "01ca309fd73f4e15ba835c39423be41d", "PLAYER_EVENT": "start", "TITLE": "Tainted", "ARTIST": "Einmusik"}
    # hitting pause in Spotify
    May 20 12:22:17 Dorayaki sh[6979]: INFO:librespot::player_event_handler: Running ["librespot.onevent"] with environment variables {"PLAYER_EVENT": "stop", "TRACK_ID": "01ca309fd73f4e15ba835c39423be41d"}
    # disconnecting from device by selecting local playback in Spotify
    May 20 12:22:23 Dorayaki sh[6979]: INFO:librespot::player_event_handler: Running ["librespot.onevent"] with environment variables {"TRACK_ID": "01ca309fd73f4e15ba835c39423be41d", "PLAYER_EVENT": "stop"}
    
So there is no difference between disconnecting and pausing. The above flag `RESTART_ON_STOP = True` and checking for it in `def onPlayBackStopped(self)` will prevent restart of the system service when called by the librespot event. Such a restart would result in the device disconnecting in the Spotify App. On the other hand, when stop is called by Kodi this behaviour is intended.

Reply:
Thank you for clarifying. Looks good to me and works here. Many thanks for adapting to Kodi hooks to use latest librespot upstream. Please update according to comments of cvh.

Reply:
@Obrekr Clarity and better features/code is always a good thing. If you think it's a fairly quick change then we can wait and roll into this PR. If you think it will take time we can merge this PR and simply bump the addon again when things are ready - there's no rush. What do you think?

Reply:
Implementing and testing the mentioned changes to librespot will take me time.
So I think this PR should be merged and I will create a new one once I think the new pause/disconnect feature is ready.